python/ray/dashboard/node_stats.py

```diff
@@ -130,8 +179,13 @@ def _update_from_actor_tasks(task, task_spec_type,
                                          invalid_state_type):
                 actor_id = ray.utils.binary_to_hex(
                     b64decode(task[task_spec_type]["actorId"]))
-                task["state"] = -1
-                task["invalidStateType"] = invalid_state_type
+                if invalid_state_type == "pendingActor":
+                    task["state"] = -1
+                elif invalid_state_type == "infeasibleActor":
+                    task["state"] = -2
+                else:
```

Comment:
Isn't this always raising errors? I think invalid state type could be just empty string or something?

Reply:
`invalid_state_type` is passed into this function. It's essentially a runtime type-check, which is not ideal, but the function isn't called currently with None or an empty string.

Reply:
Makes sense. We should eventually replace with type checking and pydantic. But this makes sense.